source/s3_auto_ranged_put.c

```diff
@@ -575,49 +663,18 @@ static int s_s3_auto_ranged_put_stream_complete(
                 return AWS_OP_ERR;
             }
 
-            s_s3_auto_ranged_put_lock_synced_data(auto_ranged_put);
-
             /* Store the multipart upload id and set that we are ready for sending parts. */
-            auto_ranged_put->synced_data.upload_id = upload_id;
-            auto_ranged_put->synced_data.state = AWS_S3_AUTO_RANGED_PUT_STATE_SENDING_PARTS;
+            auto_ranged_put->upload_id = upload_id;
 
+            s_s3_auto_ranged_put_lock_synced_data(auto_ranged_put);
+            auto_ranged_put->synced_data.state = AWS_S3_AUTO_RANGED_PUT_STATE_SENDING_PARTS;
```

Comment:
What if the state is canceling now? So, if we were in `waiting_for_create` state and user requested cancel, this complete will happen from the other thread which will overwrite the cancel, right?